example/lib/star_wars/planet_screen.dart

```diff
@@ -0,0 +1,206 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:flutter_hooks_gallery/star_wars/redux.dart';
+import 'package:flutter_hooks_gallery/star_wars/star_wars_api.dart';
+import 'package:provider/provider.dart';
+
+/// This handler will take care of async api interactions
+/// and updating the store afterwards.
+class _PlanetHandler {
+  /// constructor
+  _PlanetHandler(this._store, this._starWarsApi);
+
+  final Store<AppState, ReduxAction> _store;
+  final StarWarsApi _starWarsApi;
+
+  /// This will load all planets and will dispatch all necessary actions
+  /// on the redux store.
+  Future<void> fetchAndDispatch([String url]) async {
+    _store.dispatch(FetchPlanetPageActionStart());
+    try {
+      final page = await _starWarsApi.getPlanets(url);
+      _store.dispatch(FetchPlanetPageActionSuccess(page));
+    } catch (e) {
+      _store.dispatch(FetchPlanetPageActionError('Error loading Planets'));
+    }
+  }
+}
+
+/// This example will load, show and let you navigate through all star wars
+/// planets.
+///
+/// It will demonstrate on how to use [Provider] and [useReducer]
+class PlanetScreen extends HookWidget {
+  @override
+  Widget build(BuildContext context) {
+    final api = useMemoized(() => StarWarsApi());
+
+    final store = useReducer(
+      reducer,
+      initialState: AppState(),
```

Comment:
If possible:
```suggestion
      initialState: const AppState(),
```

Reply:
this is not working. I don't know if this is possible with built_value.

Reply:
Ah indeed, they don't have const constructors...
It's quite inefficient though.

Maybe expose a static final `AppState empty = AppState();` ?

Reply:
Since it's used only on the first build, recreating it on every build app is bad. Although it's not critical